test/dist_autograd_test.py

```diff
@@ -258,6 +285,9 @@ def _test_graph(self, fn):
             send_functions = ctx._send_functions()
             self.assertEqual(1, len(send_functions))
             self._verify_graph_for_rpc_call_exec(list(send_functions.values())[0])
+            # this barrier is needed so one worker does not clean up their
+            # autograd context before another worker tries to access it.
+            dist.barrier()
```

Comment:
@zhaojuanmao with these changes, had to add `dist.barrier()`s here and in other tests so there is no races for the autograd context.